sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -2079,4 +2078,620 @@ case class GpuConv(num: Expression, fromBase: Expression, toBase: Expression)
   override def third: Expression = toBase
 
   override def dataType: DataType = StringType
-}
\ No newline at end of file
+}
+
+case class GpuFormatNumber(x: Expression, d: Expression)
+    extends GpuBinaryExpression with ExpectsInputTypes with NullIntolerant {
+  
+  override def left: Expression = x
+  override def right: Expression = d
+  override def dataType: DataType = StringType
+  override def nullable: Boolean = true
+  override def inputTypes: Seq[AbstractDataType] = Seq(NumericType, IntegerType)
+
+  private def reduceLeadingZeros(cv: ColumnVector): ColumnVector = {
+    // replace "[0]+" to "0" in a column vector
```

Comment:
But it does not really do that, unless you are guaranteed that the input never starts with a 0. Could we have better docs about this?

Reply:
Added a comment and moved this back to `getPartsFromDecimal` because it is ad-hoc and only used once.